repos/system_upgrade/el7toel8/actors/pcidevicesscanner/libraries/pcidevicesscanner.py

```diff
@@ -0,0 +1,73 @@
+import functools
+import os
+import shlex
+import subprocess
+import six
+
+from leapp.models import PCIDevice
+
+
+def aslist(f):
+    ''' Decorator used to convert generator to list '''
+    @functools.wraps(f)
+    def inner(*args, **kwargs):
+        return list(f(*args, **kwargs))
+    return inner
+
+
+def call(args, split=True):
+    ''' Call external processes with some additional sugar '''
+    r = None
+    with open(os.devnull, mode='w') as err:
+        if six.PY3:
+            r = subprocess.check_output(args, stderr=err, encoding='utf-8')
+        else:
+            r = subprocess.check_output(args, stderr=err).decode('utf-8')
+    if split:
+        return r.splitlines()
+    return r
+
+
+def get_from_list(l, idx, default=''):
+    ''' Get item at index from list or return a default '''
+    try:
+        ret = l[idx]
+    except IndexError:
+        ret = default
+
+    return ret
+
+
+@aslist
+def get_pci_devices():
+    ''' Get all PCI devices from system '''
+    try:
+        items = call(['lspci', '-mm'])
+    except subprocess.CalledProcessError:
+        items = []
```

Comment:
How does the actor behaves when you get to this line? This function would return empty list. How does `actor.produce()` behave when you feed it with `PCIDevices(devices=[])`? It would be great to have a unit test for that. To be able to unit test it, I would move the produce to the `libraries/pcidevicesscanner.py` as well. Similar to https://github.com/oamg/leapp-repository/pull/5/files#diff-7a7cabedb8bdac2014c9c17631ff259aR24.